src/inmanta/agent/agent.py

```diff
@@ -977,16 +977,18 @@ async def _init_agent_map(self) -> None:
             self.agent_map = dict(cfg.agent_map.get())
 
     async def _init_endpoint_names(self) -> None:
-        if self.hostname is not None:
-            await self.add_end_point_name(self.hostname)
-        else:
-            # load agent names from the config file
-            agent_names = cfg.agent_names.get()
-            if agent_names is not None:
-                for name in agent_names:
-                    if "$" in name:
-                        name = name.replace("$node-name", self.node_name)
-                    await self.add_end_point_name(name)
+        assert self.agent_map is not None
+        endpoints: Iterable[str] = (
+            [self.hostname]
+            if self.hostname is not None
+            else (
+                self.agent_map.keys()
+                if cfg.use_autostart_agent_map.get()
```

Comment:
Doesn't this duplicate the logic of the method above (`_init_agent_map`)?

Reply:
I don't think so. The agent map maps endpoint names to uris, this method determines which endpoints to add. Only in the case of the autostarted agent do we take the agent map as the authority for both.

Reply:
right